### ISSUE TITLE
Cannot install Chocolatey and then MySQL package in a single Chef run.

### ISSUE THREAD

**Issue by pfaffle** [chocolatey/chocolatey-cookbook#45]
Attempting to install Chocolatey and then MySQL via the cookbook in a single Chef run results in breakage...

```
==> default: Recipe: win7mysql::default
==> default:   * chocolatey[mysql] action install
==> default:     * execute[install package mysql] action run
==> default:
==> default:
==> default:       ================================================================================
==> default:       Error executing action `run` on resource 'execute[install package mysql]'
==> default:       ================================================================================
==> default:
==> default:       Mixlib::ShellOut::ShellCommandFailed
==> default:       ------------------------------------
==> default:       Expected process to exit with [0], but received '1'
==> default:       ---- Begin output of "C:\ProgramData\chocolatey/bin/choco" install -y  mysql ----
==> default:       STDOUT: Chocolatey v0.9.9.8
==> default:
==> default:       Installing the following packages:
==> default:
==> default:       mysql
==> default:       By installing you accept licenses for the packages.
==> default:
==> default:
==> default:
==> default:       lessmsi v1.3
==> default:
==> default:        Creating  C:\ProgramData\chocolatey\lib\lessmsi\tools\AddWindowsExplorerShortcut.exe.ignore
==> default:
==> default:            Directory: C:\ProgramData\chocolatey\lib\lessmsi\tools
==> default:
==> default:        Mode                LastWriteTime     Length Name
==> default:
==> default:        ----                -------------     ------ ----
==> default:
==> default:        -a---        07/16/2015     15:38          0 AddWindowsExplorerShortcut.exe.ign
==> default:
==> default:                                                     ore
==> default:
==> default:        Downloading LessMsi 32 bit
==> default:
==> default:          from 'https://github.com/activescott/lessmsi/releases/download/v1.3/lessmsi-v1.3.zip'
==> default:
==> default:        Extracting C:\Users\TW\AppData\Local\Temp\chocolatey\LessMsi\LessMsiInstall.zip to C:\ProgramData\chocolatey\lib\lessmsi\tools...
==> default:
==> default:        C:\ProgramData\chocolatey\lib\lessmsi\tools
==> default:
==> default:        WARNING: Write-ChocolateySuccess is deprecated. If you are the maintainer,
==> default:
==> default:        please remove it from your package file.
==> default:
==> default:        ShimGen has successfully created a shim for lessmsi-gui.exe
==> default:        ShimGen has successfully created a shim for lessmsi.exe
==> default:
==> default:        The install of lessmsi was successful.
==> default:
==> default:
==> default:
==> default:       7zip.commandline v9.38
==> default:
==> default:        Downloading 7zip.commandline 32 bit
==> default:
==> default:          from 'http://www.7-zip.org/a/7z938.msi'
==> default:
==> default:        The term 'lessmsi' is not recognized as the name of a cmdlet, function, script
==> default:
==> default:        file, or operable program. Check the spelling of the name, or if a path was inc
==> default:
==> default:        luded, verify that the path is correct and try again.
==> default:
==> default:        At C:\ProgramData\chocolatey\lib\7zip.commandline\tools\chocolateyInstall.ps1:1
==> default:
==> default:        4 char:2
==> default:
==> default:        + & <<<< lessmsi x "$fileName" "$installDir\"
==> default:
==> default:            + CategoryInfo          : ObjectNotFound: (lessmsi:String) [], CommandNotF
==> default:
==> default:           oundException
==> default:
==> default:            + FullyQualifiedErrorId : CommandNotFoundException
==> default:
==> default:        Downloading  32 bit
==> default:
==> default:          from 'http://www.7-zip.org/a/7z938-extra.7z'
==> default:
==> default:        Extracting C:\Users\TW\AppData\Local\Temp\chocolatey\Install.zip to C:\ProgramData\chocolatey\lib\7zip.commandline\tools\7zip...
==> default:
==> default:        Mode                LastWriteTime     Length Name
==> default:
==> default:        ----                -------------     ------ ----
==> default:
==> default:        d----        07/16/2015     15:39            7zip
==> default:
==> default:        C:\ProgramData\chocolatey\lib\7zip.commandline\tools\7zip
==> default:
==> default:       The install of 7zip.commandline was NOT successful.
==> default:
==> default:       Error while running 'C:\ProgramData\chocolatey\lib\7zip.commandline\tools\chocolateyInstall.ps1'.
==> default:
==> default:        See log for details.
==> default:
==> default:
==> default:
==> default:       mysql v5.6.25.20150611
==> default:
==> default:        Get-BinRoot is going to be deprecated by v1. Many packages no longer require it since the folders no longer have versions on them.
==> default:
==> default:        Adding 'C:\tools\mysql\current\bin' to the path and the current shell path
==> default:
==> default:        PATH environment variable does not have C:\tools\mysql\current\bin in it. Adding...
==> default:
==> default:        Mode                LastWriteTime     Length Name
==> default:
==> default:        ----                -------------     ------ ----
==> default:
==> default:        d----        07/16/2015     15:39            mysql
==> default:
==> default:        d----        07/16/2015     15:39            mysql
==> default:
==> default:        Downloading mysql 32 bit
==> default:
==> default:          from 'http://cdn.mysql.com/Downloads/MySQL-5.6/mysql-5.6.25-win32.zip'
==> default:
==> default:        Extracting C:\Users\TW\AppData\Local\Temp\chocolatey\mysql\mysql.zip to C:\tools\mysql...
==> default:
==> default:        C:\tools\mysql
==> default:
==> default:        Shutting down MySQL if it is running
==> default:        [SC] OpenService FAILED 1060:
==> default:
==> default:        The specified service does not exist as an installed service.
==> default:
==> default:        Copying contents of 'C:\tools\mysql\mysql-5.6.25-win32' to 'C:\tools\mysql\current'.
==> default:
==> default:        d----        07/16/2015     15:42            bin
==> default:
==> default:        Installing the mysql service
==> default:
==> default:        Service successfully installed.
==> default:
==> default:        The MySQL service is starting.
==> default:
==> default:        The MySQL service was started successfully.
==> default:
==> default:        The install of mysql was successful.
==> default:
==> default:
==> default:
==> default:       Chocolatey installed 2/3 package(s). 1 package(s) failed.
==> default:
==> default:        See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
==> default:       Failures:
==> default:
==> default:        - 7zip.commandline
==> default:       STDERR:
==> default:       ---- End output of "C:\ProgramData\chocolatey/bin/choco" install -y  mysql ----
==> default:       Ran "C:\ProgramData\chocolatey/bin/choco" install -y  mysql returned 1
==> default:
==> default:       Resource Declaration:
==> default:       ---------------------
==> default:       # In C:\vagrant-chef\365bb30fc3a3f8d71c302da722515691\cookbooks\chocolatey\providers\default.rb
==> default:
==> default:       119:   execute "install package #{name}" do
==> default:       120:     command "#{::ChocolateyHelpers.chocolatey_executable} install -y #{cmd_args} #{name}"
==> default:       121:   end
==> default:       122: end
==> default:
==> default:       Compiled Resource:
==> default:       ------------------
==> default:       # Declared in C:\vagrant-chef\365bb30fc3a3f8d71c302da722515691\cookbooks\chocolatey\providers\default.rb:119:in `install'
==> default:
==> default:       execute("install package mysql") do
==> default:         action [:run]
==> default:         retries 0
==> default:         retry_delay 2
==> default:         default_guard_interpreter :execute
==> default:         command "\"C:\\ProgramData\\chocolatey/bin/choco\" install -y  mysql"
==> default:         backup 5
==> default:
==> default:         returns 0
==> default:         declared_type :execute
==> default:         cookbook_name :win7mysql
==> default:       end
==> default:
==> default: [2015-07-16T15:43:38-07:00] INFO: Running queued delayed notifications before re-raising exception
==> default:
==> default:
==> default:     ================================================================================
==> default:     Error executing action `install` on resource 'chocolatey[mysql]'
==> default:     ================================================================================
==> default:
==> default:     Mixlib::ShellOut::ShellCommandFailed
==> default:     ------------------------------------
==> default:     execute[install package mysql] (C:\vagrant-chef\365bb30fc3a3f8d71c302da722515691\cookbooks\chocolatey\providers\default.rb line 119) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
```

This chef run was launched through a Vagrant provisioning run, but I get the same behavior when I run chef-solo manually on the box as well. The recipe I wrote is very minimal, just:

```ruby
include_recipe 'chocolatey::default'
chocolatey 'mysql'
```

I suspect the problem is that the cookbook launches a separate powershell instance to download and install Chocolatey, so the PATH never gets updated in the original shell to include C:\ProgramData\chocolatey\bin, and subsequently any Chocolatey package that depends on being able to find its dependent packages in the path will fail to install.

Let me know if I can provide any additional information!

Thanks,
Craig

**Comment by ferventcoder** [comment]
@pfaffle I think you are on the right track with a subprocess updating the path - I think the fix is to update the path based on where we think choco is installing to (check it though).

**Comment by pfaffle** [comment]
Just submitted a pull request to fix this... I figured it was reasonable to just add %ChocolateyInstall%\bin to the path during the Chef run, since that's where the Chocolatey executables should be.

**Comment by mwrock** [comment]
Should be fixed by #68

**Comment by mwrock** [comment]
It "should" be working. It works for me running on a fresh 2012r2 box without Choco. Not running via scheduled task.

Could you provide the error you are getting with debug logging?  Are you running the task under a normal user account or the system account?

**Comment by noresjo** [comment]
Thanks for answering.

 I've tried some divide and conquer on the problem and I'm now pretty sure that this is related to Scheduled task. The setup I am using is basically: Terraform creates the machine and does basic provisioning via winrm and bootstraps a chef client. The chef client starts on system startup to have reboot resiliency (a must on lovely Windows). Chef then installs chocolatey and chocolatey packages.

If I install chocolaty/packages during basic provisioning via winrm everything works fine.

When I do both install of chocolatey and packages a Scheduled task it does not work. Even if I reboot. The scheduled task starts on system startup.  (see below for error message). 
 - This used to work a couple of months ago (choco 0.9.x)
 - I can still login to the machine via rdp or winrm and run choco.exe. Then it works.

Command line used for to create the task:
`SchTasks /Create /SC ONSTART /RL HIGHEST /TN chef-client /RU foo /RP bar /F /TR "C:\\opscode\\chef\\bin\\chef-client.bat -L c:\\temp\log.txt`

Use foo is local admin.

```
---- Begin output of "C:\ProgramData\chocolatey/bin/choco" install -y  notepadplusplus ----
STDOUT: Chocolatey v0.10.0

Installing the following packages:

notepadplusplus

By installing you accept licenses for the packages.



notepadplusplus.install v6.9.2 [Approved]

Downloading notepadplusplus 
  from 'https://notepad-plus-plus.org/repository/6.x/6.9.2/npp.6.9.2.Installer.exe'


Progress: 3% - Saving 143.61 KB of 4.02 MB (147052/4211112)
Progress: 7% - Saving 303.61 KB of 4.02 MB (310892/4211112)
Progress: 11% - Saving 463.61 KB of 4.02 MB (474732/4211112)
Progress: 15% - Saving 623.61 KB of 4.02 MB (638572/4211112)
Progress: 19% - Saving 783.61 KB of 4.02 MB (802412/4211112)
Progress: 22% - Saving 943.61 KB of 4.02 MB (966252/4211112)
Progress: 26% - Saving 1.08 MB of 4.02 MB (1130092/4211112)
Progress: 30% - Saving 1.23 MB of 4.02 MB (1293932/4211112)
Progress: 34% - Saving 1.39 MB of 4.02 MB (1457772/4211112)
Progress: 38% - Saving 1.55 MB of 4.02 MB (1621612/4211112)
Progress: 42% - Saving 1.7 MB of 4.02 MB (1785452/4211112)
Progress: 46% - Saving 1.86 MB of 4.02 MB (1949292/4211112)
Progress: 50% - Saving 2.02 MB of 4.02 MB (2113132/4211112)
Progress: 54% - Saving 2.17 MB of 4.02 MB (2276972/4211112)
Progress: 57% - Saving 2.33 MB of 4.02 MB (2440812/4211112)
Progress: 61% - Saving 2.48 MB of 4.02 MB (2604652/4211112)
Progress: 65% - Saving 2.64 MB of 4.02 MB (2768492/4211112)
Progress: 69% - Saving 2.8 MB of 4.02 MB (2932332/4211112)
Progress: 73% - Saving 2.95 MB of 4.02 MB (3096172/4211112)
Progress: 77% - Saving 3.11 MB of 4.02 MB (3260012/4211112)
Progress: 81% - Saving 3.27 MB of 4.02 MB (3423852/4211112)
Progress: 85% - Saving 3.42 MB of 4.02 MB (3587692/4211112)
Progress: 89% - Saving 3.58 MB of 4.02 MB (3751532/4211112)
Progress: 92% - Saving 3.73 MB of 4.02 MB (3915372/4211112)
Progress: 96% - Saving 3.89 MB of 4.02 MB (4079212/4211112)
Progress: 100% - Saving 4.02 MB of 4.02 MB (4211112/4211112)
Progress: 100% - Completed download of C:\Windows\TEMP\chocolatey\notepadplusplus.install\6.9.2\npp.6.9.2.Installer.exe (4.02 MB).

Download of npp.6.9.2.Installer.exe (4.02 MB) completed.

Installing notepadplusplus...

notepadplusplus has been installed.

 The install of notepadplusplus.install was successful.

  Software installed as 'exe', install location is likely default.



notepadplusplus v6.9.2 [Approved]

 The install of notepadplusplus was successful.

  Software install location not explicitly set, could be in package or 

  default install location if installer.

notepadplusplus not installed. An error occurred during installation:

 Path to combine cannot be empty.

The install of notepadplusplus was NOT successful.

notepadplusplus not installed. An error occurred during installation:

 Path to combine cannot be empty.



Chocolatey installed 1/2 packages. 1 packages failed.

 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).



Failures

 - notepadplusplus (exited 1) - notepadplusplus not installed. An error occurred during installation:

 Path to combine cannot be empty.
STDERR: 
---- End output of "C:\ProgramData\chocolatey/bin/choco" install -y  notepadplusplus ----
Ran "C:\ProgramData\chocolatey/bin/choco" install -y  notepadplusplus returned 1

```

`Path to combine cannot be empty.` seems to be the problem here. I have a feeling this is related to chocolatey installation environment paths.  



Also, if I install chocolatey via win rm, then create the scheduled task, then reboot and it attempt to install choco packages via the schedueled task(chef), it still does not work. Strange to me.


Here is the end of the chocolatey log:

```
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyZipPackage'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Install-Vsix'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Set-EnvironmentVariable'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Set-PowerShellExitCode'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Start-ChocolateyProcessAsAdmin'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Test-ProcessAdminRights'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Uninstall-BinFile'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Uninstall-ChocolateyEnvironmentVariable'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Uninstall-ChocolateyPackage'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Uninstall-ChocolateyZipPackage'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Update-SessionEnvironment'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Write-ChocolateyFailure'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Write-ChocolateySuccess'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing function 'Write-FileUpdateLog'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing alias 'Add-BinFile'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing alias 'Generate-BinFile'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing alias 'Get-InstallRegistryKey'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing alias 'Get-OSBitness'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing alias 'Get-ProcessorBits'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing alias 'refreshenv'.
2016-08-25 23:10:02,960 [INFO ] - VERBOSE: Importing alias 'Remove-BinFile'.
2016-08-25 23:10:02,960 [DEBUG] - Running 'ChocolateyScriptRunner' for notepadplusplus v6.9.2 with packageScript 'C:\ProgramData\chocolatey\lib\notepadplusplus\tools\chocolateyInstall.ps1', packageFolder:'C:\ProgramData\chocolatey\lib\notepadplusplus', installArguments: '', packageParameters: '',
2016-08-25 23:10:02,960 [DEBUG] - Running 'C:\ProgramData\chocolatey\lib\notepadplusplus\tools\chocolateyInstall.ps1'
2016-08-25 23:10:02,976 [DEBUG] - Built-in PowerShell host called with ['[System.Threading.Thread]::CurrentThread.CurrentCulture = '';[System.Threading.Thread]::CurrentThread.CurrentUICulture = ''; & import-module -name 'C:\ProgramData\chocolatey\helpers\chocolateyInstaller.psm1'; & 'C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1' -packageScript 'C:\ProgramData\chocolatey\lib\notepadplusplus\tools\chocolateyInstall.ps1' -installArguments '' -packageParameters '''] exited with '0'.
2016-08-25 23:10:02,976 [DEBUG] - Calling command ['"C:\Windows\System32\shutdown.exe" /a']
2016-08-25 23:10:02,991 [DEBUG] - Command ['"C:\Windows\System32\shutdown.exe" /a'] exited with '1116'
2016-08-25 23:10:02,991 [DEBUG] - Capturing package files in 'C:\ProgramData\chocolatey\lib\notepadplusplus'
2016-08-25 23:10:02,991 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\notepadplusplus\.chocolateyPending'
  with checksum '1E445B2F134A2A3ECAE85682B852626F'
2016-08-25 23:10:02,991 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\notepadplusplus\notepadplusplus.nupkg'
  with checksum 'E6E9608F2D431C1DF44A01501D62D53D'
2016-08-25 23:10:02,991 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\notepadplusplus\notepadplusplus.nuspec'
  with checksum 'E1A6294C7CE24B157A74C6FFF569589D'
2016-08-25 23:10:02,991 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\notepadplusplus\tools\chocolateyInstall.ps1'
  with checksum 'E681F355000DAF7C0D94E48772AF8A38'
2016-08-25 23:10:02,991 [DEBUG] - Attempting to create directory "C:\ProgramData\chocolatey\.chocolatey\notepadplusplus.6.9.2".
2016-08-25 23:10:02,991 [DEBUG] - Attempting to copy "C:\ProgramData\chocolatey\.chocolatey\notepadplusplus.6.9.2\.files.update"
 to "C:\ProgramData\chocolatey\.chocolatey\notepadplusplus.6.9.2\.files".
2016-08-25 23:10:02,991 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\notepadplusplus.6.9.2\.files.update".
2016-08-25 23:10:02,991 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\notepadplusplus.6.9.2\.sxs".
2016-08-25 23:10:02,991 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\notepadplusplus.6.9.2\.pin".
2016-08-25 23:10:02,991 [DEBUG] - Sending message 'HandlePackageResultCompletedMessage' out if there are subscribers...
2016-08-25 23:10:02,991 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\lib\notepadplusplus\.chocolateyPending".
2016-08-25 23:10:02,991 [INFO ] -  The install of notepadplusplus was successful.
2016-08-25 23:10:02,991 [INFO ] -   Software install location not explicitly set, could be in package or 
  default install location if installer.
2016-08-25 23:10:02,991 [ERROR] - notepadplusplus not installed. An error occurred during installation:
 Path to combine cannot be empty.
2016-08-25 23:10:03,007 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\notepadplusplus.6.9.2\.files.update".
2016-08-25 23:10:03,007 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\notepadplusplus.6.9.2\.arguments".
2016-08-25 23:10:03,007 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\notepadplusplus.6.9.2\.sxs".
2016-08-25 23:10:03,007 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\notepadplusplus.6.9.2\.pin".
2016-08-25 23:10:03,007 [DEBUG] - Sending message 'HandlePackageResultCompletedMessage' out if there are subscribers...
2016-08-25 23:10:03,007 [ERROR] - The install of notepadplusplus was NOT successful.
2016-08-25 23:10:03,023 [ERROR] - notepadplusplus not installed. An error occurred during installation:
 Path to combine cannot be empty.
2016-08-25 23:10:03,023 [DEBUG] - Attempting to create directory "C:\ProgramData\chocolatey\lib-bad".
2016-08-25 23:10:03,023 [DEBUG] - Moving 'C:\ProgramData\chocolatey\lib\notepadplusplus'
 to 'C:\ProgramData\chocolatey\lib-bad\notepadplusplus'
2016-08-25 23:10:05,082 [WARN ] - 
Chocolatey installed 1/2 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
2016-08-25 23:10:05,082 [INFO ] - 
2016-08-25 23:10:05,082 [ERROR] - Failures
2016-08-25 23:10:05,082 [ERROR] -  - notepadplusplus (exited 1) - notepadplusplus not installed. An error occurred during installation:
 Path to combine cannot be empty.
2016-08-25 23:10:05,082 [DEBUG] - Sending message 'PostRunMessage' out if there are subscribers...
2016-08-25 23:10:05,097 [DEBUG] - Exiting with 1
```

**Comment by mwrock** [comment]
Thanks for the details @noresjo. They are very helpful and this is a completely different error than I was anticipating. @ferventcoder have you seen anything like this when rubbing choco inside a scheduled task? One thing that might be interesting @noresjo is to have a scheduled task call a .bat that just does the `choco install` instead of going through `chef-client`. That would help to isolate whether this is chef related at all or not.

**Comment by ferventcoder** [comment]
> when rubbing choco inside a scheduled task

I can't say I've ever run into doing this. ;)

**Comment by noresjo** [comment]
Hi again. Indeed, just doing `choco install -y terraform` in a .bat file as a scheduled task does not work. Seems that chef is not the problem here. Unless the fact that chocolatey is installed through chef as well. 

Here's the output:
```
Chocolatey v0.10.0
[Pending] Removing incomplete install for 'terraform'
Installing the following packages:
terraform
By installing you accept licenses for the packages.

terraform v0.7.1 [Approved]
Removing old terraform plugins
Downloading terraform 64 bit
  from 'https://releases.hashicorp.com/terraform/0.7.1/terraform_0.7.1_windows_amd64.zip'

Progress: 0% - Saving 26.78 KB of 15.52 MB (27420/16270466)
Progress: 1% - Saving 160 KB of 15.52 MB (163840/16270466)
...
Progress: 99% - Saving 15.47 MB of 15.52 MB (16220160/16270466)
Progress: 100% - Completed download of C:\Windows\TEMP\chocolatey\terraform\0.7.1\terraform_0.7.1_windows_amd64.zip (15.52 MB).
Download of terraform_0.7.1_windows_amd64.zip (15.52 MB) completed.
Extracting C:\Windows\TEMP\chocolatey\terraform\0.7.1\terraform_0.7.1_windows_amd64.zip to C:\ProgramData\chocolatey\lib\terraform\tools...
C:\ProgramData\chocolatey\lib\terraform\tools
 ShimGen has successfully created a shim for terraform.exe
 The install of terraform was successful.
  Software installed to 'C:\ProgramData\chocolatey\lib\terraform\tools'
terraform not installed. An error occurred during installation:
 Path to combine cannot be empty.
The install of terraform was NOT successful.
terraform not installed. An error occurred during installation:
 Path to combine cannot be empty.
```

**Comment by noresjo** [comment]
I also tried to install chocolatey via the standard way
`@powershell -NoProfile -ExecutionPolicy Bypass -Command "iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))" && SET "PATH=%PATH%;%ALLUSERSPROFILE%\chocolatey\bin"` in cmd.  Then the same scheduled task as above with a simple choco install. Same error. Does not work.

**Comment by ferventcoder** [comment]
@noresjo can you provide the log file for this? I want to determine if it is a bug in Chocolatey or a bug in the installation script. The log file will have debug/verbose output, so scrub it for sensitive data.

We will need the entire set of output though. If you are not comfortable putting it in a gist and linking that, feel free to send it to me at ferventcoder [at] gmail. Thanks!

**Comment by ferventcoder** [comment]
I'm seeing something similar in notepadplusplus in your logs above and would love to get to the bottom of this. We've taken steps in 0.10.0 to address empty path combines.

**Comment by noresjo** [comment]
I've sent the chocolatey.log file to the two of you.

Another interesting thing I noticed... I create the task and it is set to run on system startup. This is how I need it to be.. but if I then (just for testing) tell the task to run immediately in the current session. it actually works. Very weird.

**Comment by noresjo** [comment]
I have also tried run the scheduled task as "SYSTEM". Exactly the same behavior.

**Comment by noresjo** [comment]
Should I maybe create a separate issue for this?

**Comment by mwrock** [comment]
Yeah an issue on the choco repo makes sense.

**Comment by noresjo** [comment]
I'll do it when I can find the time. I downgraded to choco "0.9.9.11" since I can now do that with this cookbook. Everything seems to work fine now.

**Comment by ferventcoder** [comment]
This could be a similar issue - https://github.com/chocolatey/choco/issues/956

**Comment by ferventcoder** [comment]
This will be fixed for 0.10.1 thanks to @noresjo - see https://github.com/chocolatey/choco/issues/945 for details!
